File: 2124.检查是否所有A都在B之前.py
#
# @lc app=leetcode.cn id=2124 lang=python3
#
# [2124] 检查是否所有 A 都在 B 之前
#

# @lc code=start
class Solution:
    def checkString(self, s: str) -> bool:
        a_status = False
        b_status = False
        for i in range(len(s)):
            if s[i] == "a":
                if b_status:
                    return False
                if not a_status:
                    a_status = True
            else:
                if not b_status:
                    b_status = True
        return True

# @lc code=end

File: 2124.检查是否所有A都在B之前.go
/*
 * @lc app=leetcode.cn id=2124 lang=golang
 *
 * [2124] 检查是否所有 A 都在 B 之前
 */

// @lc code=start
func checkString(s string) bool {
	aStatus := false
	bStatus := false
	a := byte('a')
	// 遍历，如果 a 出现时 b 已经出现（即 bStatus 为 true），则返回 false
	for i := 0; i < len(s); i++ {
		if s[i] == a {
			if !aStatus {
				aStatus = true
			}
			if bStatus {
				return false
			}
		} else {
			if !bStatus {
				bStatus = true
			}
		}
	}
	return true

}

// @lc code=end

